src/Lean/Server/InfoUtils.lean

```diff
@@ -86,6 +86,27 @@ where go ctx? a
     ts.foldl (init := a) (go <| i.updateContext? ctx?)
   | _ => a
 
+/--
+Fold an info tree as follows, while ensuring that the correct `ContextInfo` is supplied at each stage:
+
+ * Nodes are combined with the initial value `init` using `f`, and the result is then combined with the children using a left fold
+ * Everything else returns the initial value
```

Comment:
It may be a good idea to be explicit about "everything else" (perhaps in the code as well). The remaining case are holes.

Reply:
Done.